tests/test_radiolist.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import unicode_literals
+
+from prompt_toolkit.widgets import RadioList
+
+import pytest
+
+def test_initial():
+    values = [
+                (1, 'some_text'),
+                ('fizz', 'some_more_text'),
+                ({'foo':'bar'}, 'even_more_text')
+             ]
+    radiolist = RadioList(values)
+    assert radiolist.current_value == 1
+    assert radiolist._selected_index == 0
+
+def test_default():
+    values = [
+                (1, 'some_text'),
+                ('fizz', 'some_more_text'),
+                ({'foo':'bar'}, 'even_more_text')
+             ]
+    radiolist = RadioList(values, 1)
+    assert radiolist.current_value == 'fizz'
+    assert radiolist._selected_index == 1
+
+def test_bad_params():
+    with pytest.raises(AssertionError):
```

Comment:
Hi @Arrowbox,
Thanks a lot for writing unit tests, but for these, we shouldn't ever assume that ``AssertionError`` is ever raised. If Python runs with the ``-O`` option, then no assertions are ever raised. Removing an assertion from the code is not supposed to break something, so there can't be a unit test which states it does.

I think raising ``ValueError`` is more appropriate, if we want to have some flow control in case of passing a bad input.

Reply:
I'm happy to change that, but there are a bunch of other `assert precondition` statements at the beginning of the function already. I was trying to follow the previous format and thought an extra test there would be okay. Should I change all the checks? Or maybe just drop the test?

Reply:
I looked at testing if assert has been stripped to decide if we should just drop the test but that did not seem trivial.